<a href="https://colab.research.google.com/github/kevincruzortiz123-lgtm/programacion-2/blob/main/parcial2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Kevin Jhovany Cruz Ortiz
parte II
instruccion 1

In [114]:
import numpy as np
np.random.seed(0)
datos = np.random.randint(108, 197, 50)
b=np.array([datos])

In [109]:
a=np.unique(datos)
a


array([117, 120, 122, 127, 128, 129, 131, 133, 137, 139, 140, 143, 144,
       145, 147, 152, 154, 155, 157, 165, 166, 172, 173, 175, 177, 178,
       180, 182, 183, 185, 187, 188, 189, 190, 191, 195, 196])

In [115]:
import pandas as pd


normalizar = lambda x: (b - b.mean()) / b.std()
resultado = normalizar(b)
print(resultado)

[[-0.30985234 -0.19307382  0.46867113  0.58544966  0.58544966 -1.67226843
   1.20826844 -1.20515435 -0.62126174  1.36397314  0.70222818  1.40289931
   1.40289931 -1.55548991  0.23511409  0.50759731 -0.50448321  1.36397314
  -0.232       1.40289931  1.13041609 -0.58233556 -1.04944965  0.9747114
   0.78008053 -1.67226843 -1.24408052  1.09148992  0.663302    1.05256374
  -0.19307382  0.46867113  1.16934227  1.40289931 -0.11522148 -0.89374495
  -1.28300669 -1.28300669 -1.47763756 -0.50448321 -0.77696643  0.50759731
  -1.67226843  0.19618792 -0.77696643 -0.81589261  0.85793287 -1.127302
  -0.66018791  0.89685905]]


In [111]:
import pandas as pd
datos1=pd.DataFrame(datos)
datos1.describe()

,0
count,50.000000
mean,159.960000
std,25.950471
min,117.000000
25%,139.250000
50%,161.000000
75%,182.750000
max,196.000000


el maximo valor es de 196 y el de menor de 117

In [99]:
from google.colab import files
import csv
datos=[ np.array(["datos"]),
        b

]
with open("datos.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerows(datos)

files.download("datos.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

instruccion 2

In [47]:
df= pd.DataFrame({
    "nombre":["pepe","juan","roberto","maria"],
    "edad":[32,20,21,36],
    "carrera":["matematicas","quimica","biologia","medicina"],
    "ciudad":["bogota","medellin","bucaramanga","cundimarca"],
    "nota":[4.0, 2.3, 1.9, 3.4]
})
df


,nombre,edad,carrera,ciudad,nota
0,pepe,32,matematicas,bogota,4.0
1,juan,20,quimica,medellin,2.3
2,roberto,21,biologia,bucaramanga,1.9
3,maria,36,medicina,cundimarca,3.4


In [48]:
df["ciudad"]

,ciudad
0,bogota
1,medellin
2,bucaramanga
3,cundimarca


In [49]:
df.head(2)

,nombre,edad,carrera,ciudad,nota
0,pepe,32,matematicas,bogota,4.0
1,juan,20,quimica,medellin,2.3


In [50]:
df.loc[1]

,1
nombre,juan
edad,20
carrera,quimica
ciudad,medellin
nota,2.3


In [51]:
df[df["nota"]<3.0]

,nombre,edad,carrera,ciudad,nota
1,juan,20,quimica,medellin,2.3
2,roberto,21,biologia,bucaramanga,1.9


In [52]:
df["nota"].apply(lambda x: x+0.3)

,nota
0,4.3
1,2.6
2,2.2
3,3.7


In [57]:
# Agrupar por nombre y nota
grupo = df.groupby("nombre")["nota"].mean()
grupo


,nota
nombre,
juan,2.3
maria,3.4
pepe,4.0
roberto,1.9


In [135]:
from google.colab import files
import csv
datos=[
    ["nombre","pepe","juan","roberto","maria"],
    ["edad",32,20,21,36],
    ["carrera","matematicas","quimica","biologia","medicina"],
    ["ciudad","bogota","medellin","bucaramanga","cundimarca"],
    ["nota",4.0, 2.3, 1.9, 3.4]


]
with open("datos.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerows(datos)

files.download("datos.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Parte III

In [62]:
np.random.seed(0)
prueba_A= np.random.normal(loc=7.2, scale=0.4,size=52)
prueba_B= np.random.gamma(shape=306, scale=0.0229, size=52)
examen=pd.DataFrame({
    "prueba_A":prueba_A,
    "prueba_B":prueba_B
})


In [67]:
examen.max()

,0
prueba_A,8.107902
prueba_B,7.774417


In [68]:
examen.min()

,0
prueba_A,6.178804
prueba_B,6.331116


In [69]:
examen.describe()

,prueba_A,prueba_B
count,52.000000,52.000000
mean,7.250149,7.020939
std,0.449705,0.348294
min,6.178804,6.331116
25%,6.972888,6.748714
50%,7.259798,7.059122
75%,7.554340,7.261352
max,8.107902,7.774417


In [213]:
from scipy import stats

In [149]:
from scipy.stats import shapiro
shapiro(examen["prueba_A"],)

ShapiroResult(statistic=np.float64(0.9885919443153024), pvalue=np.float64(0.8970354954225455))

como el valor p es mayor que 0.05, no se rechaza la hipotesis de normalidad con una confianza del 95%, entonces se cumple e supuesto de normalidad

In [150]:
shapiro(examen["prueba_B"])

ShapiroResult(statistic=np.float64(0.9808232976972334), pvalue=np.float64(0.5614479966043984))

como el valor p es mayor que 0.05, no se rechaza la hipotesis de normalidad con una confianza del 95%, entonces se cumple e supuesto de normalidad

In [82]:
t_stat, p_val = stats.ttest_rel(examen["prueba_A"], examen["prueba_B"])
print("Estadístico t:", t_stat)
print("Valor p:", p_val)


Estadístico t: 2.768664456056009
Valor p: 0.007827648453462765


No se rechaza que el puntaje medio en ambas encuestas es igual con una confianza del 95%

In [166]:
stats.kendalltau

<function scipy.stats._stats_py.kendalltau(x, y, *, nan_policy='propagate', method='auto', variant='b', alternative='two-sided', axis=None, keepdims=False)>

No se rechaza con una confianza del 95% que los puntajes de ambas encuestas no están correlacionados

In [172]:
stats.spearmanr

<function scipy.stats._stats_py.spearmanr(a, b=None, axis=0, nan_policy='propagate', alternative='two-sided')>

No se rechaza con una confianza del 95% que los puntajes de ambas encuestas no están correlacionados

In [152]:
import os
os.makedirs("comparacion", exist_ok=True)

In [219]:
%%writefile comparacion/pruebat.py
def prueba(df):
  from scipy import stats
  t_stat, p_val = stats.ttest_rel(df.loc["prueba_A"], df.loc["prueba_B"])
  print("Estadístico t:", t_stat)
  print("Valor p:", p_val)

Overwriting comparacion/pruebat.py


In [208]:
%%writefile comparacion/cor.py
def prueba1(df):
  from scipy import stats
  stats.kendalltau
  stats.spearmanr


Writing comparacion/cor.py


In [222]:
df = pd.DataFrame({
"Estudiante": range(1, 11),
"Prueba_A": [7.1, 7.5, 6.8, 7.3, 7.0, 7.4, 6.9, 7.2, 7.6, 7.1],
"Prueba_B": [6.9, 7.2, 7.0, 7.1, 6.8, 7.3, 7.0, 7.1, 7.4, 7.0]
})
df

,Estudiante,Prueba_A,Prueba_B
0,1,7.1,6.9
1,2,7.5,7.2
2,3,6.8,7.0
3,4,7.3,7.1
4,5,7.0,6.8
5,6,7.4,7.3
6,7,6.9,7.0
7,8,7.2,7.1
8,9,7.6,7.4
9,10,7.1,7.0


# CREAR UN CSV:
from google.colab import files
import csv
datos = [
    ["Nombre", "Edad", "Ciudad"],
    ["Ana", 25, "Bogotá"],
    ["Luis", 30, "Medellín"],
    ["Carlos", 28, "Cali"]
]

with open("datos.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerows(datos)

files.download("datos.csv")

# CREAR MODULOS Y PAQUETES:

import os
os.makedirs("op", exist_ok=True)

%%writefile op/operaciones.py

# FUNCION LAMBDA
import pandas as pd
x = [1, 2, 3, 4]
x = pd.DataFrame(x)
normalizar = lambda x: (x - x.mean()) / x.std()
resultado = normalizar(x)
print(resultado)
